docs/design/PicoD-Plain-Authentication-Design.md

```diff
@@ -0,0 +1,218 @@
+# Picod Plain Authentication Design
+
+Author: Layne Peng
+
+## Motivation
+
+Currently, AgentCube’s `picod` daemon enforces authentication and authorization using a client self-signed key-pair mechanism. This design binds a specific client (agent) directly to a `picod` instance, ensuring a secure, one-to-one relationship. Details of this existing implementation can be found in the [PicoD Authentication & Authorization Design](https://github.com/volcano-sh/agentcube/blob/main/docs/design/picod-proposal.md#3-authentication--authorization).
+
+However, emerging use cases require a more flexible architecture where the client's authentication and authorization are offloaded to an upstream Router or Gateway. In this scenario, the Router is responsible for:
+*   Allocating the appropriate built-in service (PicoD) to the user.
+*   Managing the security of connections between clients (agents) and the service.
+
+The existing self-signed key-pair model is incompatible with this centralized management flow, as it bypasses the Router's ability to mediate access. To address this, we propose a new **Plain Authentication** mechanism for `picod`. This design enables the Router/Gateway to manage credentials and connection security, simplifying the client-side workflow while maintaining robust access control.
+## Use Cases
+
+### Gateway-Managed Sandbox Access
+
+A user requests a sandbox environment using the AgentCube Python SDK. The Router receives this request and validates the user's identity (authorization logic is handled upstream). Upon successful validation, the Router selects and allocates an available PicoD instance. It records the mapping between the specific Session ID and the requesting client. The Router then returns the connection credentials to the client, enabling the Python SDK to establish a direct connection with the PicoD instance using the plain authentication mechanism.
+
+## Design Details
+
+### Architecture Overview
+
+To ensure **High Availability (HA)** across multiple Router replicas and enforce the principle of **Least Privilege**, this design implements a **Shared Identity Model** backed by Kubernetes primitives.
+
+1. **Shared Authority (Router)**:
+    
+    - All Router replicas share a single cryptographic identity to function as a unified Certificate Authority (CA).
+    - **Private Key Storage**: Stored in a Kubernetes Secret (picod-router-identity). This is restricted to the Router namespace/service account.
+    - **Public Key Distribution**: Published to a Kubernetes ConfigMap (picod-router-public-key). This is accessible by the WorkloadManager and PicoD instances.
```

Comment:
I donot think this is a good practice to make workload manager startup depend on `router`.  In workload manager we can use downward api to project the public key to the code interpreter without reading the key